Lib/Src/CParser.cpp

```diff
@@ -0,0 +1,309 @@
+/**
+ * @file CParser.cpp
+ */
+/*
+ * Created on : Apr 4, 2022
+ * Author : niuslar
+ */
+
+#include "CParser.h"
+
+#define ERROR_VALUE (0.00)
+
+/**
+ * @brief Constructor
+ */
+CParser::CParser() {}
+
+/**
+ * @brief Parse string
+ * @param String to be parsed
+ * @return Last parsing state
+ */
+CParser::parser_state_t CParser::parse(
+    const etl::string<MAX_STRING_SIZE> &string)
+{
+    // Reset command variables
+    m_status = IDLE;
+    m_argument_counter = 0;
+    m_command_name = "";
+
+    // Get tokens from string
+    getTokens(string);
+
+    for (uint16_t i = 0; i < m_token_counter; i++)
+    {
+        token_t current_token = m_tokens[i];
+
+        switch (current_token.type)
```

Comment:
Have you considered building state machine based on parser state, not current token? The way you did it is rather unconventional.

Reply:
This state machine is very hard to follow. The state changes depending on the value of the token, but your implemented it as if token change depending on state.